go.mod

```diff
@@ -25,7 +25,7 @@ require (
 	k8s.io/client-go v12.0.0+incompatible
 	k8s.io/utils v0.0.0-20211116205334-6203023598ed
 	sigs.k8s.io/controller-runtime v0.11.2
-	sigs.k8s.io/controller-tools v0.4.1
+	sigs.k8s.io/controller-tools v0.8.0
```

Comment:
The latest version is 0.9.2 - any reason we don't go to that?

Reply:
Because the `operator-sdk` framework by default used 0.8.0 controller tools version when I initialised the operator in a [separate PR on my fork](https://github.com/Jaanki/submariner-operator/pull/10/files?file-filters%5B%5D=.mod&file-filters%5B%5D=.yaml&file-filters%5B%5D=No+extension&show-deleted-files=true&show-viewed-files=true#diff-3bf1b6c02ff8069cea7b4c42751006ce5baafde160d68e6272782f726e9c738bR6). I am trying to replicate the code changes from forked PR to PRs here.

Reply:
I just tried with `0.9.2` and `make manifests` fails with below errors

```
mkdir -p /go/src/github.com/submariner-io/submariner-operator/bin
go build -o /go/src/github.com/submariner-io/submariner-operator/bin/controller-gen sigs.k8s.io/controller-tools/cmd/controller-gen
# sigs.k8s.io/controller-tools/pkg/crd/markers
vendor/sigs.k8s.io/controller-tools/pkg/crd/markers/validation.go:490:9: schema.XValidations undefined (type *"k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1".JSONSchemaProps has no field or method XValidations)
vendor/sigs.k8s.io/controller-tools/pkg/crd/markers/validation.go:490:38: schema.XValidations undefined (type *"k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1".JSONSchemaProps has no field or method XValidations)
vendor/sigs.k8s.io/controller-tools/pkg/crd/markers/validation.go:490:59: undefined: v1.ValidationRule
make: *** [Makefile:160: /go/src/github.com/submariner-io/submariner-operator/bin/controller-gen] Error 2
make: *** [Makefile.dapper:50: manifests] Error 2
```

Reply:
OK we'll need to bump the K8s versions as well (eg `k8s.io/apiextensions-apiserver`) to go to _0.9.2_.  I don't know which `operator-sdk` version you're bumping to but I imagine it would be the latest _v1.22.2_ which bumps `controller-tools` to [v0.9.0](https://github.com/operator-framework/operator-sdk/blob/da3346113a8a75e11225f586482934000504a60f/go.mod#L44) (and K8s to _0.24.0_). But we can do that later. We've bumped every other submariner project to K8s _0.24.2_ so we we'll want **submariner-operator** to follow suit.

Reply:
Given that we'll want to bump `controller-tools` to _0.9.*_ anyway, I wonder how useful this PR is standalone, ie you'll have to run `make manifests` again anyway..